crates/typst/src/eval/ops.rs

```diff
@@ -500,54 +513,96 @@ pub fn equal(lhs: &Value, rhs: &Value) -> bool {
         // Some technically different things should compare equal.
         (&Int(a), &Float(b)) => a as f64 == b,
         (&Float(a), &Int(b)) => a == b as f64,
-        (&Length(a), &Relative(b)) => a == b.abs && b.rel.is_zero(),
+        (&Length(a), &Relative(b)) => {
```

Comment:
nit: let's combine these pairs of `(&Foo(a), &Bar(b))` and `(&Bar(b), &Foo(a))` as something like `commutative!(&Foo(a), &Bar(b))`? So that you don't have to write the similar code twice.

```rs
macro_rules! commutative {
  ($a:pat, $b:pat) => { ($a, $b) | ($b, $a) };
}
```

Reply:
This has been covered by https://github.com/typst/typst/pull/3720, PR needs rebasing.

Reply:
I see! I still selfishly prefer the macros now that we are at it :)